discovery_engine_core/bindings/src/tracing.rs

```diff
@@ -14,37 +14,105 @@
 
 //! Setup tracing on different platform.
 
-use std::sync::Once;
+use std::{env::VarError, error::Error, fs::OpenOptions, io, path::Path, sync::Once};
 
-use tracing_subscriber::{filter::LevelFilter, util::SubscriberInitExt};
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter};
 
 static INIT_TRACING: Once = Once::new();
 
-pub(crate) fn init_tracing() {
+pub(crate) fn init_tracing(log_file: Option<&Path>) {
     INIT_TRACING.call_once(|| {
-        init_tracing_once();
+        init_tracing_once(log_file);
+        init_panic_logging();
     });
 }
 
-fn init_tracing_once() {
-    let subscriber = tracing_subscriber::fmt()
-        .with_max_level(LevelFilter::INFO)
-        .finish();
+fn init_tracing_once(log_file: Option<&Path>) {
+    let mut delayed_errors = Vec::new();
 
+    let stdout_log = tracing_subscriber::fmt::layer();
+
+    let android_logging;
     cfg_if::cfg_if! {
         if #[cfg(target_os = "android")] {
-            use tracing_subscriber::layer::SubscriberExt;
-            let layer = tracing_android::layer("xayn_discovery_engine").ok();
-            subscriber.with(layer).init();
+            android_logging = tracing_android::layer("xayn_discovery_engine").ok()
```

Comment:
It makes no sense to add this error to delayed errors as:

- we most likely won't see the log in that case anyway
- it can only fail if the hard coded tag string  is changed to contain '\0'